back/app/controllers/web_api/v1/ideas_controller.rb

```diff
@@ -21,19 +21,24 @@ def index
     ideas = IdeasFinder.new(
       params,
       scope: policy_scope(Idea).where(publication_status: 'published'),
-      current_user: current_user,
-      includes: [
-        :idea_images, :idea_trending_info, :topics,
-        :idea_import, # defined through BulkImportIdeas engine
-        {
-          project: [:phases, { custom_form: [:custom_fields] }],
-          phases: [:permissions],
-          author: [:unread_notifications]
-        }
-      ]
+      current_user: current_user
     ).find_records
+
     ideas = paginate SortByParamsService.new.sort_ideas(ideas, params, current_user)
 
+    # Only include after pagination - so we only get associations we need
+    ideas = ideas.includes(
+      :idea_images,
+      :idea_trending_info,
+      :topics,
+      :idea_import, # defined through BulkImportIdeas engine
+      {
+        project: [:phases, { custom_form: [:custom_fields] }],
+        phases: [:permissions],
+        author: [:unread_notifications]
+      }
+    )
```

Comment:
Whilst analysing queries I realised that in the projects and ideas controller we were including/preloading far more than we needed to (ie we were preloading for all ideas and not just the ones on this page). Anecdotally when I changed this for a project I have locally with 250 ideas - the load time for the page more than halved.

Reply:
I've also gone through all the other controllers to make sure that `includes` always follows `paginate`